### ISSUE TITLE
Possibility to only mount a subdirectory?

### ISSUE THREAD

**Issue by 50m30n3** [odeke-em/drive#761]
I've looked through the documentation but didn't see this addressed anywhere.
It would be really useful if it was possible have a way to only mount a subdirectory from Google Drive.

For example I would like to sync my `~/Documents` directory with a corresponding `/Documents` directory in Google Drive. 
But right now it seems like I can only mount the entire drive and then selectively sync the `/Documents` directory, by doing `$ drive pull Documents` but of course that syncs it into a new subdirectory.
I could mount the drive to my home directory, but that just seems like a bad idea.

I also tried mounting the drive to `~/GDrive` and then symlinking my `~/Documents` directory to `~/GDrive/Documents` but calling `$ drive pull` in the symlinked directory doesn't work since it doesn't traverse the symlink.

Also I'd really like to avoid mounting my entire Google Drive, since it has tons of data in it that I don't want to sync to my computer.

Basically, it would be really nice to be able to specify a subdirectory when initializing the drive, something like `$ drive init --subdirectory=/Documents ~/Documents`, and then everything you do happens relative to this subdirectory.

**Comment by odeke-em** [comment]
Hello there @50m30n3, thank you the question and welcome to drive!

You can pull in only a subset of your drive e.g
```shell
drive pull GDrive/Documents/a1/d2
``` 

without having to pull in the entire drive. That's one of the mandates of this project, to be able to pull in selective directories and files without having to sync everything. In fact, on my computers, I only have very small subsets of my folders at a time, to conserve my disk space.

You can also pull to a desired depth, only pull folders, only pull files, see https://github.com/odeke-em/drive#pulling.

Please try that and let me know what you think.

**Comment by 50m30n3** [comment]
Hi @odeke-em!

I'm aware of that functionality, so let me try to rephrase what I mean.

I'm basically looking for a way to sync an existing directory somewhere in my file system to a sub-directory in my drive.
I would like to, for example, sync `/sync/foo/` on my drive to `~/somedir/foo/` and maybe `/sync/bar` to `~/baz/`.

My idea would be to have something like

`$ drive init --subdirectory=/sync/bar ~/baz`

that would init the `~/bar` directory and basically use `/sync/bar` on my drive as the "root" for that directory.

I think this feature would be really useful for people like me who don't want to move all their files into a central directory to sync them.

**Comment by wyldfury** [comment]
Hi,

I second this idea. In my case I want various dotfiles in my homedir synced to a subdir on Google Drive. To achieve this at the moment I have to init, for example, /drive/mydrive and then have /home/user be a link to /drive/mydrive/homedirs/user. Otherwise I'd be cluttering the root of my Google Drive with my dotfiles.

If it were possible to init to a specific subfolder on my GDrive then I could just init my homedir directly to that subfolder and be done.

I do see some potential problems though. Like would it be possible to init multiple local dirs to different GDrive subdirs? May not be a problem, but I don't know how the credentials work on the GDrive side.
Another scenario I see is, what happens if you init a local directory to one GDrive account and then init a subdir of the local dir to a different GDrive account? Just add the subdir to .driveignore on the parent directory?

Anyway, just some thoughts on the idea.

Cheers

**Comment by mscheper** [comment]
I would seriously love this as well. The way I work around this with DropBox is that I have a ~/DropBox directory, and I symlink subdirectories to where I want them in my directory tree. I found this thread when I searched to see whether I need to do the same for `drive`.

It's a reasonable workaround, but it runs into the classic problems with symlinks in backup systems, and of course it's not an option under Windows (which I thankfully don't use very often, but I can't avoid it completely). So adding support for this to `drive` would be a big win.

(I've actually never understood why so few cloud storage systems have this feature. It seems a basic requirement, especially for Google Drive, which dumps shared resources in the root of its directory structure. Is file organisation really an egghead power-user thing these days?! Ironically, DropSync, the tool I use for syncing my Android to DropBox, has this feature, even though Android apps frustratingly try to hide the directory system.)
